mne_connectivity/base.py

```diff
@@ -645,6 +650,16 @@ def n_epochs_used(self):
         """
         return self.attrs.get('n_epochs_used')
 
+    @property
+    def rank(self):
+        """Seed and target ranks used in the connectivity."""
+        return self.attrs['rank']
+
+    @property
+    def n_lags(self):
+        """Number of lags used in the model for computing connectivity."""
+        return self.attrs['n_lags']
+
```

Comment:
For these additional properties + `patterns`, I would be inclined to put these altogether into a new class for `MultivariateMixin`.

Slash, do we necessarily need these? That is we don't need to expose all `attrs`. Rn it exposes the core ones, but presumably a user can store arbitrary data fields in there.

Reply:
I only included them here so that they would be included in the docs for the connectivity classes.
Something like `patterns` I would be more inclined to expose, since these are relevant results of the class, however I see no issue not exposing `rank` and `n_lags`.
Perhaps I could write a `MultivariateMixin` for `patterns`, or do you think it would be a bit overkill to only expose one attribute?

Reply:
`patterns` are the spatial topographies I mentioned in the previous PR (generated from the spatial filters used to compute MIC). They can also be called spatial patterns, and this "patterns" terminology is more consistent with other parts of MNE, e.g. [mne.decoding.SSD](https://mne.tools/dev/generated/mne.decoding.SSD.html).

Reply:
Ah I see.

If they are only used for `MIC`, then I would just document the `MIC` function stating that `patterns` is available in the `attrs` object. Perhaps even explicitly showing this in the example should suffice too. Otherwise it would be weird to have a property always be set to `None` unless MIC is used.